test/extended/operators/certs.go

```diff
@@ -49,6 +56,82 @@ var _ = g.Describe("[sig-arch][Late]", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 
+	g.It("all tls artifacts must be known", func() {
+
+		ctx := context.Background()
+		kubeClient := oc.AdminKubeClient()
+		if ok, _ := exutil.IsMicroShiftCluster(kubeClient); ok {
+			g.Skip("microshift does not auto-collect TLS.")
+		}
+
+		nodes := map[string]int{}
+		controlPlaneLabel := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/control-plane": ""})
+		nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: controlPlaneLabel.String()})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i, node := range nodeList.Items {
+			nodes[node.Name] = i
+		}
+
+		actualPKIContent, err := certgraphanalysis.GatherCertsFromPlatformNamespaces(ctx, kubeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		expectedPKIContent, err := certs.GetPKIInfoFromEmbeddedOwnership()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		messages := []string{}
+
+		unregisteredSecrets := map[certgraphapi.InClusterSecretLocation]certgraphapi.PKIRegistryInClusterCertKeyPair{}
+
+		for _, currCertKeyPair := range actualPKIContent.InClusterResourceData.CertKeyPairs {
+			currLocation := currCertKeyPair.SecretLocation
+			expectedSecret, err := certgraphutils.LocateCertKeyPair(currLocation, expectedPKIContent.CertKeyPairs)
+			if err != nil {
+				unregisteredSecrets[currLocation] = currCertKeyPair
```

Comment:
once our expected no longer includes revisioned secrets, this will be vulnerable.  Refactoring as suggested https://github.com/openshift/origin/pull/28305/files#r1376642834 will bring them both into the same type as `*certgraphapi.PKIRegistryInfo` which we likely want.